hummingbot/strategy/pure_market_making/pure_market_making.pyx

```diff
@@ -705,10 +707,16 @@ cdef class PureMarketMakingStrategy(StrategyBase):
             list new_sells = []
         top_ask = market.c_get_price(self.trading_pair, True)
         if not top_ask.is_nan():
-            proposal.buys = [buy for buy in proposal.buys if buy.price < top_ask]
+            if not self._take_if_crossed:
```

Comment:
It'd be simpler to just not call `c_filter_out_takers` if `take_if_crossed` is set.
i.e. c_tick
```
if not self._take_if_crossed:
   self.c_filter_out_takers(proposal)
```

Reply:
The call to `c_filter_out_takers` is still very necessary because there is need for the bot to submit a matching limit taker order on the current exchange with **a price that matches** existing maker orders. This is necessary in order to prevent the price on the current exchange(if it is a less liquid market) from converging to the price of the external source.

Kindly look through #1680 for better explanation.

Reply:
I had taken a look at the requirement before giving my review. The way I see it is if the order price crosses to the other side, let it matches and fills. What you did is to limit the order price to only the top bid/ask, if the top bid/ask does not have enough liquidity, the rest unfilled amount will be left as a maker on the book. 

Note that exchanges match orders by best price possible first. so an order with limit price over top bid/ask still gets filled at the top bid/ask first.

Reply:
Taking the top bid/ask and letting the extra rest as a maker was the way I saw it. I thought that would be the desired effect, just like you interpreted.

However, I get your point.
The reason I didn't implement as you see it is to prevent a great dislocation in a less liquid market and to reduce the rate of converging to the price of the external price source for a high liquid market.

Should I still go on to implement it as you see it or do you have any plans to clarify first?

Reply:
From what I understand, the main reason to set this parameter on is to use PMM also as arbitrage, users believe that the price on the PMM exchange is dislocated, as such turning maker into taker is justified. By how much is the dislocation? The simplest way is to use users bid and ask spread setting (could also be an extra parameter for this, but I think it'll get too complicated). By taking only the top bid/ask, we assume that only the top bid/ask is dislocated, also what happens when those orders are dust orders? 
But I also see your point, by converting the crosses over orders to take on only the top bid/ask makes the bot less intrusive to the market and somewhat safer mode of operation.

Reply:
I think we should go with @Nullably's interpretation of this issue, since I think that's the original intent of the request. It's possible that it might cause more dislocation in the market and weird behavior but we we won't find out unless we allow users to experiment with it.